tests/lanczos_simpleop.hpp

```diff
@@ -0,0 +1,75 @@
+/*************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_LANCZOS_SIMPLEOP_HPP
+#define MFMG_LANCZOS_SIMPLEOP_HPP
+
+#include <mfmg/common/exceptions.hpp>
+#include <mfmg/common/operator.hpp>
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+namespace mfmg
+{
+
+//-----------------------------------------------------------------------------
+/// \brief Simple test operator
+///
+///        A diagonal matrix with equally spaced eigenvalues of some
+///        multiplicity.
+
+template <typename VectorType_>
+class SimpleOperator
+{
+public:
+  // Typedefs
+  using VectorType = VectorType_;
+  using ScalarType = typename VectorType::value_type;
+
+  typedef typename std::vector<VectorType *> Vectors_t;
+
+  // Ctor/dtor
+
+  SimpleOperator(size_t dim, size_t multiplicity = 1);
+  ~SimpleOperator();
+
+  // Accessors
+  ScalarType eigenvalue(size_t i) const { return diag_value_(i); }
+
+  // Operations
+
+  void vmult(VectorType &y, VectorType const &x) const;
+
+  size_t m() const { return _dim; }
+  size_t n() const { return _dim; }
+
+private:
+  size_t _dim;
+  size_t _multiplicity;
+
+  // This will be a diagonal matrix; specify the diag entries here
+
+  ScalarType diag_value_(size_t i) const
```

Comment:
I don't see the point of that member function.  Why don't you just move the definition into `eigenvalue()`.

Reply:
Because diagonal value is also used in `apply`.

Reply:
So?  `SimpleOperator::vmult()` could call directly `eigenvalue()`.   I just didn't see the point of having two member functions but I suppose it's a way of declaring an alias of the function and I can see that either names is more suitable depending on the context.  I guess I can live with it :)